### PR TITLE
"Count" option for secret-revealed message

### PR DESCRIPTION
**Pull request by MrAlaux**

### CODE REVIEW

File: src/mn_setup.c [hunk 1]
```diff
@@ -1952,8 +1953,13 @@ static setup_menu_t stat_settings3[] = {
     MI_END
 };
 
+static const char *secretmessage_strings[] = {
+    "Off", "Simple", "Count",
```
Comment (fabiangreffrath) [inline src/mn_setup.c:1957]: Hm, I don't like the term "simple". Any other ideas?

Comment (rfomin) [reply src/mn_setup.c:1957]: How about removing the "Simple" option? So "Count" would be just "On".

Comment (fabiangreffrath) [reply src/mn_setup.c:1957]: So, we *always* count revealed secrets?

Comment (rfomin) [reply src/mn_setup.c:1957]: > So, we always count revealed secrets?

Yes, but maybe some users won't like it.

Comment (fabiangreffrath) [reply src/mn_setup.c:1957]: I'm sure some people wouldn't like it. How about "Off", "Report", "Count"?

Comment (MrAlaux) [reply src/mn_setup.c:1957]: FWIW, ever since I implemented the feature in Nugget months (years?) ago, it's been _Off_, _On_ and _Count_.

Comment (fabiangreffrath) [reply src/mn_setup.c:1957]: It's the same in Crispy, actually.

File: src/p_spec.c [hunk 2]
```diff
@@ -2174,7 +2174,18 @@ static void P_SecretRevealed(player_t *player)
 {
   if (hud_secret_message && player == &players[consoleplayer])
   {
-    player->secretmessage = s_HUSTR_SECRETFOUND;
+    if (hud_secret_message == SECRETMESSAGE_COUNT)
+    {
+      static char str_count[32];
+      M_snprintf(str_count, sizeof(str_count), "Secret %d of %d revealed!", player->secretcount, totalsecret);
```
Comment (fabiangreffrath) [inline src/p_spec.c:2180]: Could you break this line, please?

File: src/mn_setup.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/mn_setup.c:1957]: So, could you change it to "Off", "On" and "Count", please?

Comment (MrAlaux) [reply src/mn_setup.c]: Done in https://github.com/fabiangreffrath/woof/pull/1966/commits/ea80bf1cd7eafd5c8cd86a7d2ff1c9653148f712.

File: src/p_spec.c (hunk 2 above)
Comment (MrAlaux) [reply src/p_spec.c]: Done in https://github.com/fabiangreffrath/woof/pull/1966/commits/ea80bf1cd7eafd5c8cd86a7d2ff1c9653148f712.

**Comment by MrAlaux** [comment]
Any suggestions, @rfomin? If not, I'll merge it.

**Review by rfomin — APPROVED** [review]
Looks good, thank you.

**Comment by MrAlaux** [comment]
Oh, I'm a fool: the CVAR description still reads "1 = Simple". Is it intended or did you miss it too?

**Comment by rfomin** [comment]
> Is it intended or did you miss it too?

I missed it.

**Comment by fabiangreffrath** [comment]
I missed it, too. But it's not a big deal.

**Comment by MrAlaux** [comment]
> Oh, I'm a fool: the CVAR description still reads "1 = Simple".

For posterity: fixed in https://github.com/fabiangreffrath/woof/commit/09a1315ea8b072c56f423844f814adda748dd3e4.

**Pull request merged**